include/seqan3/core/configuration/detail/concept.hpp

```diff
@@ -183,3 +181,51 @@ SEQAN3_CONCEPT config_element_pipeable_with =
 //!\endcond
 
 } // namespace seqan3::detail
+
+namespace seqan3
+{
+//!\cond
+// Forward declaration.
+template <detail::config_element ... configs_t>
+class configuration;
+//!\endcond
+
+/*!\brief Helper variable template to test if a configuration element is combineable with another configuration element
+ *        or configuration.
+ * \ingroup algorithm
+ *
+ * \tparam config1_t Either the type of a configuration element or a configuration.
+ * \tparam config2_t Either the type of a configuration element or a configuration.
+ *
+ * \details
+ *
+ * This helper variable template checks if `config1_t` fulfills the concept requirements
+ * seqan3::detail::config_element_pipeable_with `config2_t`. If `config2_t` is a seqan3::configuration, the check will
+ * be expanded to every configuration element contained in the configuration type. Only if `config1_t` is combineable
+ * with every element stored inside of the given configuration, this helper variable template evaluates to `true`,
+ * otherwise `false`.
+ * If `config1_t` is a seqan3::configuration the same applies in combination with the configuration element `config2_t`.
+ * If both `config1_t` and `config2_t` are seqan3::configuration types, then the cartesian product between the
+ * configuration elements of the first configuration and the second configuration are tested.
+ */
+template <typename config1_t, typename config2_t>
+inline constexpr bool is_config_element_combineable_v = detail::config_element_pipeable_with<config1_t, config2_t>;
```

Comment:
Considering our stable / experimental discussions, does this need to be seqan3::is_config_element_combineable_v ?

Reply:
Well, configuration is at least right now supposed to be stable. That means that an error that uses entities that are not in user scope as well are hard to track for the user. So it seems reasonable to offer this in seqan3:: namespace. Though, it is  not something a user would necessarily use in it's own code base. Would that be something NOAPI would then be good for?

Reply:
Interesting idea. Maybe NOAPI is in this case like exposition-only in the standard. We define the intent, but don't fully specify it. :)

Reply:
I like that idea, can you add \noapi to this?

Reply:
@rrahn I did that for you :)